File: words/words.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() 
{ 
	char *buffer = malloc(sizeof(char)*100);
    char *token = malloc(sizeof(char)*100);
  	if (buffer == NULL || token == NULL) {
    	printf("Error allocating memory");
      	exit(1);
    }
  	int wordCount = 0;
      fgets(buffer, 512, stdin);
      if (buffer != NULL) {
        while ((token = strtok_r(buffer, " \n\t", &buffer))) {
          wordCount++;
        }
    }
  printf("Your text contains %d words\n", wordCount);
  return 0;
} 

File: words/words.rs
use std::io;

fn main() {
    println!("Enter your text :");

    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read the line");

    let words: Vec<&str> = input.trim().split_whitespace().collect();
    println!("Your text contains {} words.", words.len());
}